UE plugin documentation update (#8164)
getsentry/sentry-docs: src/platforms/unreal/configuration/setup-crashreporter.mdx

</Alert>

## UE Crash reporter

The UE Crash Reporter is provided out of the box by Epic Games. It is able to collect relevant
information about the crash and send it to a service like Sentry to process.

Adding it to your game means that in case a crash happens, the following dialog is displayed
to the user:

![The UE Crash Reporter](ue-crash-reporter.png)

### Include the UE Crash Reporter

You can add the _Crash Reporter Client_ to your game in your _Project Settings_.

The simplest way is to search the option: `Crash Reporter`:

![Include UE Crash Reporter](ue-include-crash-reporter.png)

The option is located under _Project > Packaging_ menu, then select _show advanced_ followed by
checking the box for: `Include Crash Reporter`.

## Debug information

To get the most out of captured crash reports [debug symbols need to be uploaded to Sentry](#upload-debug-symbols).
In order for Sentry to be able to process the crash report and translate
memory addresses to meaningful information like function names, module names
and line numbers, the crash itself must be associated with the corresponding debug information first.

## Configure the Crash Reporter Endpoint

> Now that the _Crash Reporter_ and _Debug Files_ are included, UE needs to know where to send the
> crash. For that, we add the Sentry _Unreal Engine Endpoint_ from the _Client Keys_ settings page to game's configuration file. This will
> include which project within Sentry you want to see the crashes arriving in real time.
> That's accomplished by configuring the `CrashReportClient` in the _DefaultEngine.ini_ file. Changing the engine is necessary for this to work.

Reviewer: ```suggestion
Now that the _Crash Reporter_ and _Debug Files_ are included, UE needs to know where to send the
crash. For that, we need to add the Sentry _Unreal Engine Endpoint_ from the _Client Keys_ settings page to the game's configuration file. This will include whichever project within Sentry you want to see real-time crashes for. That's accomplished by configuring the `CrashReportClient` in the _DefaultEngine.ini_ file. You'll need to change the engine to do this to work.
```

---

Installation of a Sentry SDK is not required in order to capture the crashes of your
Unreal Engine application or game, because Sentry supports the _UE Crash Reporter_.

To integrate your UE game or application with Sentry, the following steps are required:

1. Include the _UE Crash Reporter_ in your game or application.
2. Include Debug information in the crash reports.
3. Add the Unreal Engine Endpoint to the relevant configuration file.
4. Upload your games' symbols so Sentry can display function names and line numbers.
5. Optionally, enable Event Attachments for your Sentry project.

Below we'll break down each step in detail.

## UE Crash Reporter

The UE Crash Reporter is provided out of the box by Epic Games. It is able to collect relevant
information about the crash and send it to a service like Sentry to process.

Adding it to your game means that in case a crash happens, the following dialog is displayed
to the user:

![The UE Crash Reporter](ue-crash-reporter.png)

### Using the UE Crash Reporter on Windows

> The UE Crash Reporter works if the Sentry SDK is installed. It can help with configuring the UE Crash Reporter, uploading debug symbols, and enriching crash reports with custom data. However, it's important to keep in mind that **starting from UE 5.2** users have to choose between the two available crash capturing mechanisms - the UE Crash Reporter or the `sentry-native` library that's integrated into the Sentry plugin. By default, the second option is enabled and the UE Crash Reporter is suppressed. (See "Enable automatic crash capturing (Windows, UE 5.2+)" in the plugin settings.) The two solutions are mutually exclusive and can't be used simultaneously.

Author: > What does "It" refer to in this sentence?

> I would have guessed "It" refers to the UE Crash Reporter, but it doesn't make sense to say that "the UE Crash Reporter can help with configuring the UE Crash Reporter..."

This sentence was a bit unclear indeed - I've replaced "It" with "Sentry SDK" to avoid possible misreading here.

You need to call the `ConfigureCrashReporter` some time after your game
starts. Any [event attribute](https://develop.sentry.dev/sdk/event-payloads/) can be set.

<Note>

> Configuring Crash Reporter attributes should be completed before initializing Sentry SDK if one is enabled. Otherwise, some information could be overwritten and lost.

Reviewer: ```suggestion
You should configure Crash Reporter attributes before initializing the Sentry SDK. Otherwise, some information may get overwritten and lost.
```

</Note>

<Note>

> Sentry SDK provides API allowing to set Crash Reporter attributes as well.

Reviewer: ```suggestion
The Sentry SDK provides an API which allows you to set Crash Reporter attributes as well.
```

(sometimes also referred to as _Debug Symbols_ or just _Symbols_). We recommend
uploading debug information during your build or release process.

<Note>

> Sentry SDK can hadle [debug symbols uploading](/platforms/unreal/configuration/debug-symbols/) automatically at build time.

Reviewer: ```suggestion
The Sentry SDK can handle uploading [debug symbols](/platforms/unreal/configuration/debug-symbols/) automatically at build time.
```

## Size Limits

> Event ingestion imposes limits on the size of UE crash reports. These limits

Reviewer: ```suggestion
Event ingestion imposes size limits on UE crash reports. These limits
```

are subject to future change and defined currently as:

- _20MB_ for a compressed request
- _100MB_ for the full crash report after decompression

> ## Issues With The Crash Reporter

Reviewer: ```suggestion
## Issues With the Crash Reporter
```

> For tips on troubleshooting the UE Crash Reporter, [check out this thread in Sentry's forum](https://forum.sentry.io/t/unreal-engine-crash-reporter-cant-get-it-to-work/7643) where community members have delved into potential solutions.

Reviewer: ```suggestion
Check out this community-generated [thread in Sentry's forum](https://forum.sentry.io/t/unreal-engine-crash-reporter-cant-get-it-to-work/7643) for UE Crash Reporter troubleshooting tips and solutions.
```
